test/collection_test.js

```diff
@@ -884,35 +884,49 @@ describe("Collection", () => {
         .should.eventually.become(result);
     });
 
-    it("should transfer the headers option", () => {
-      var pullChanges = sandbox.stub(articles, "pullChanges").returns(Promise.resolve({}));
-      return articles.sync({headers: {Foo: "Bar"}})
-        .then(() => {
-          expect(pullChanges.firstCall.args[1]).eql({headers: {Foo: "Bar"}});
-        })
+    it("should resolve early on push failure", () => {
+      const pullResult = new SyncResultObject();
+      const pushResult = new SyncResultObject();
+      pushResult.add("conflicts", [1]);
+      sandbox.stub(articles, "pullChanges").returns(Promise.resolve(pullResult));
+      sandbox.stub(articles, "pushChanges").returns(Promise.resolve(pushResult));
+      return articles.sync()
+        .should.eventually.become(pushResult);
```

Comment:
nit: maybe we could change the test title to something like «pull is not executed if push fails». Just to help understanding this assertion :)